### PR TITLE
Small fixes for mongo 4.4 collections

### PR DESCRIPTION
**Pull request by wallyworld**
Remove some old mongo 2.x and 3.x compatibility code.
Also, tweak how existence of collections is checked to make things work on mongo 4.4 snap.
Some 4.0 code (flagged with a todo) can be deleted once we swap to mongo 4.4

## QA steps

unit tests
bootstrap

### CODE REVIEW

**Comment by wallyworld** [comment]
[mongo 4.4 collection fixes](https://trello.com/c/g9LyIc6o/1372-mongo-44-collection-fixes)

File: state/database.go [hunk 1]
```diff
@@ -220,10 +219,11 @@ func (schema CollectionSchema) Create(
 // createCollection swallows collection-already-exists errors.
 func createCollection(raw *mgo.Collection, spec *mgo.CollectionInfo) error {
 	err := raw.Create(spec)
-	// The lack of error code for this error was reported upstream:
-	//     https://jira.mongodb.org/browse/SERVER-6992
-	if err == nil || strings.HasSuffix(err.Error(), "already exists") {
-		return nil
+	if err, ok := err.(*mgo.QueryError); ok {
+		// 48 is collection already exists.
+		if err.Code == 48 {
```
Comment (jameinel) [inline state/database.go:224]: Should we define this as a const instead of just 48 here?

File: state/logs.go [hunk 2]
```diff
@@ -962,20 +965,13 @@ func getCollectionCappedInfo(coll *mgo.Collection) (bool, int, error) {
 	if err != nil {
 		return false, 0, errors.Trace(err)
 	}
-	// For mongo 2.4, the "capped" key is only available when it is set
-	// to true. For mongo 3.2 and 3.6, it is always there.
-	// If capped is there, but isn't a bool, we treat this as false.
-	capped, _ := stats["capped"].(bool)
+	capped := stats["capped"].(bool)
```
Comment (jameinel) [inline state/logs.go:968]: doing this without the , ok will panic if it doesn't cast correctly.
I'd rather at least keep that check in place.

**Review by jameinel — APPROVED** [review]

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**